sloglint.go

```diff
@@ -397,6 +414,56 @@ func badKeyNames(info *types.Info, caseFn func(string) string, keys, attrs []ast
 	return false
 }
 
+// checkForbiddenKeys returns the first forbidden key found in keys or attrs or an empty string.
+func checkForbiddenKeys(info *types.Info, forbiddenKeys []string, keys, attrs []ast.Expr) (string, bool) {
+	if len(forbiddenKeys) == 0 {
+		return "", false
+	}
+	for _, key := range keys {
+		if name, ok := getKeyName(key); ok && slices.Contains(forbiddenKeys, name) {
+			return name, true
+		}
+	}
+
+	for _, attr := range attrs {
+		var expr ast.Expr
+
+		switch attr := attr.(type) {
+		case *ast.CallExpr: // e.g. slog.Int()
+			fn := typeutil.StaticCallee(info, attr)
+			if fn == nil {
+				continue
+			}
+			if _, ok := attrFuncs[fn.FullName()]; !ok {
+				continue
+			}
+			expr = attr.Args[0]
+
+		case *ast.CompositeLit: // slog.Attr{}
+			switch len(attr.Elts) {
+			case 1: // slog.Attr{Key: ...} | slog.Attr{Value: ...}
+				if kv := attr.Elts[0].(*ast.KeyValueExpr); kv.Key.(*ast.Ident).Name == "Key" {
+					expr = kv.Value
+				}
+			case 2: // slog.Attr{..., ...} | slog.Attr{Key: ..., Value: ...}
+				expr = attr.Elts[0]
+				if kv1, ok := attr.Elts[0].(*ast.KeyValueExpr); ok && kv1.Key.(*ast.Ident).Name == "Key" {
+					expr = kv1.Value
+				}
+				if kv2, ok := attr.Elts[1].(*ast.KeyValueExpr); ok && kv2.Key.(*ast.Ident).Name == "Key" {
+					expr = kv2.Value
+				}
+			}
```

Comment:
Would it be possible (and practical) do this iteratively on the off-chance an extra field gets added to slog.Attr in the future? It would cause all tests for this to fail if that were to happen, but might be a nice edge case. For example:

```suggestion
			for i := range attr.Elts {
				kv, ok := attr.Elts[i].(*ast.KeyValueExpr)
				if !ok {
					continue
				}
				
				if ident, ok := kv.Key.(*ast.Ident); ok && ident.Name == "Key" {
					expr = kv.Value
					break
				}
			}
```

Either way, it might be beneficial to avoid potential panics during the casts.